src/asynk/async_queue.rs

```diff
@@ -599,32 +610,40 @@ where
         &mut self,
         task_type: Option<String>,
     ) -> Result<Option<Task>, AsyncQueueError> {
-        let mut connection = self.pool.get().await?;
-        let mut transaction = connection.transaction().await?;
+        if self.connected {
```

Comment:
instead of wrapping in `if-else`. let's create a guard function like:

```rust
check_if_connection?();
```

Reply:
Solve in [20e4757](https://github.com/ayrat555/fang/pull/54/commits/20e4757e6412d6819109d60b1ab308ded0863332)